spyder/plugins/layout/plugin.py

```diff
@@ -0,0 +1,792 @@
+        screen_shape = QApplication.desktop().geometry()
+        current_width = screen_shape.width()
+        current_height = screen_shape.height()
+        if current_width < width or current_height < height:
+            pos = get_func(prefix + 'position', section)
+
+        is_maximized = get_func(prefix + 'is_maximized', section=section)
+        is_fullscreen = get_func(prefix + 'is_fullscreen', section=section)
+        return (hexstate, window_size, prefs_dialog_size, pos, is_maximized,
+                is_fullscreen)
+
+    def get_window_settings(self):
+        """
+        Return current window settings.
+
+        Symetric to the 'set_window_settings' setter.
+        """
+        # FIXME: Window size in main window is update on resize
+        window_size = (self.window_size.width(), self.window_size.height())
+
+        is_fullscreen = self.main.isFullScreen()
+        if is_fullscreen:
+            is_maximized = self._maximized_flag
+        else:
+            is_maximized = self.main.isMaximized()
+
+        pos = (self.window_position.x(), self.window_position.y())
+        prefs_dialog_size = (self.prefs_dialog_size.width(),
+                             self.prefs_dialog_size.height())
+
+        hexstate = qbytearray_to_str(
+            self.main.saveState(version=WINDOW_STATE_VERSION))
+        return (hexstate, window_size, prefs_dialog_size, pos, is_maximized,
+                is_fullscreen)
+
+    def set_window_settings(self, hexstate, window_size, prefs_dialog_size,
+                            pos, is_maximized, is_fullscreen):
+        """
+        Set window settings Symetric to the 'get_window_settings' accessor.
+        """
+        main = self.main
+        main.setUpdatesEnabled(False)
+        self.prefs_dialog_size = QSize(prefs_dialog_size[0],
+                                       prefs_dialog_size[1])  # width,height
+        main.set_prefs_size(self.prefs_dialog_size)
+        self.window_size = QSize(window_size[0],
+                                 window_size[1])  # width, height
+        self.window_position = QPoint(pos[0], pos[1])  # x,y
+        main.setWindowState(Qt.WindowNoState)
+        main.resize(self.window_size)
+        main.move(self.window_position)
+
+        # Window layout
+        if hexstate:
+            hexstate_valid = self.main.restoreState(
+                QByteArray().fromHex(str(hexstate).encode('utf-8')),
+                version=WINDOW_STATE_VERSION)
+
+            # Check layout validity. Spyder 4 and below uses the version 0
+            # state (default), whereas Spyder 5 will use version 1 state.
+            # For more info see the version argument for
+            # QMainWindow.restoreState:
+            # https://doc.qt.io/qt-5/qmainwindow.html#restoreState
+            if not hexstate_valid:
+                self.main.setUpdatesEnabled(True)
+                self.setup_layout(default=True)
+                return
+
+            # FIXME:

```

Comment:
Why this FIXME? Do you plan to address it?

Reply:
I think that was before I started checking this so I'm unsure

Reply:
Should I remove the `FIXME`?

Reply:
Yes, please do.